parser/src/parser.rs

```diff
@@ -636,6 +636,77 @@ class Foo(A, B):
         insta::assert_debug_snapshot!(ast::Suite::parse(source, "<test>").unwrap());
     }
 
+    #[test]
+    #[cfg(feature = "all-nodes-with-ranges")]
+    fn test_parse_class_with_generic_type() {
+        let source = "\
+class Foo[T](A, B):
+  pass
+";
```

Comment:
Would it be preferable to have all of these test cases collapsed into `test_parse_class`?